pkg/csi/service/vanilla/controller_helper.go

```diff
@@ -167,12 +179,15 @@ func getBlockVolumeToHostMap(ctx context.Context, cMgr *common.Manager,
 		vmRefs = append(vmRefs, nodeVM.Reference())
 		vmRefToUUID[vmRef] = nodeVM.UUID
 	}
-	pc := property.DefaultCollector(vc.Client.Client)
-	// Obtain host MoID and virtual disk ID
-	err = pc.Retrieve(ctx, vmRefs, properties, &vmMoList)
-	if err != nil {
-		log.Errorf("failed to get VM managed objects from VM objects, err: %v", err)
-		return volumeIDNodeUUIDMap, err
+
+	for _, vc := range vCenters {
+		pc := property.DefaultCollector(vc.Client.Client)
+		// Obtain host MoID and virtual disk ID
+		err = pc.Retrieve(ctx, vmRefs, properties, &vmMoList)
```

Comment:
you can calling `pc.Retrieve` in a loop. Can you check if it is overriding the result or appending new result in the vmMoList

Reply:
verified that pc.Retrieve appends the result every time to dst pointer provided as argument, which is vmMoList here